### PR TITLE
4 additional unit-tests

### PR DESCRIPTION
**Pull request by AlexEMG**

### CODE REVIEW

File: tests/test_frame_selection_tools.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    # Index bounds: (0, 1) => valid_frame_index_1 == frame_index_1
+    frame_index_1 = list(range(100, 500, 5))
+    # Index bounds: 0.25 * 1600 = 400, 0.3 * 1600 = 480
+    frame_index_2 = list(range(100, 500, 5))
+    valid_frame_index_2 = [idx for idx in frame_index_2 if 400 <= idx <= 480]
+    for fps, duration, n_to_pick, start, end, index, valid_frame_indices in [
+        (32, 10, 10, 0, 1, None, list(range(0, 320))),  # 320 frames at 32 fps -> 10 seconds
+        (16, 100, 50, 0, 1, frame_index_1, frame_index_1),  # 1600 frames at 16 fps -> 100 seconds
+        (16, 100, 5, 0.25, 0.3, frame_index_2, valid_frame_index_2),  # 1600 frames at 16 fps -> 100 seconds
```
Comment (jeylau) [inline tests/test_frame_selection_tools.py]: Can these multiple sets of parameters be moved outside the func in @pytest.mark.parametrize()?; e.g., as in https://github.com/DeepLabCut/DeepLabCut/blob/9fcf1f8ad9f04f2e23da74a3a9dc1463ee831c8f/tests/test_video.py#L76
When a unit test fails, this makes debugging easier.

File: tests/test_frame_selection_tools.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    # Index bounds: (0, 1) => valid_frame_index_1 == frame_index_1
+    frame_index_1 = list(range(100, 500, 5))
+    # Index bounds: 0.25 * 1600 = 400, 0.3 * 1600 = 480
+    frame_index_2 = list(range(100, 500, 5))
+    valid_frame_index_2 = [idx for idx in frame_index_2 if 400 <= idx <= 480]
+    for fps, nframes, n_to_pick, start, end, index, valid_frame_indices in [
+        (32, 320, 10, 0, 1, None, list(range(0, 320))),  # 320 frames at 32 fps -> 10 seconds
+        (16, 1600, 50, 0, 1, frame_index_1, frame_index_1),  # 1600 frames at 16 fps -> 100 seconds
+        (16, 1600, 5, 0.25, 0.3, frame_index_2, valid_frame_index_2),  # 1600 frames at 16 fps -> 100 seconds
```
Comment (jeylau) [inline tests/test_frame_selection_tools.py]: Same here.

File: tests/test_auxiliaryfunctions.py [hunk 3]
```diff
@@ -119,3 +120,43 @@ def _create_fake_file(filename):
...
+
+def test_intersection_of_body_parts_and_ones_given_by_user():
+    all_bodyparts = ["head", "shoulders", "knees", "and toes", "knees and toes"]
+    comparison_bodyparts = {"shoulders", "knees and toes", "others"}
+    expected_bodyparts = ["shoulders", "knees and toes"]
+
+    for multianimal, bodyparts, multianimalbodyparts, uniquebodyparts in [
+        (False, all_bodyparts, None, None),
+        (True, None, all_bodyparts[:3], all_bodyparts[3:]),
+    ]:
```
Comment (jeylau) [inline tests/test_auxiliaryfunctions.py]: Same as below: let's use the `parametrize` fixture.

**Review by jeylau — CHANGES_REQUESTED** [review]

**Comment by jeylau** [comment]
Awesome, just left minor comments.

**Review by AlexEMG — COMMENTED** [review]
Great!

**Pull request merged**